File: car-showroom/src/main/java/com/zuzex/carshowroom/exception/GlobalExceptionHandler.java
package com.zuzex.carshowroom.exception;

import com.zuzex.common.dto.ErrorMessageDto;
import com.zuzex.common.exception.NotFoundException;
import lombok.extern.slf4j.Slf4j;
import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.ExceptionHandler;
import org.springframework.web.bind.annotation.RestControllerAdvice;

@Slf4j
@RestControllerAdvice
public class GlobalExceptionHandler {

    @ExceptionHandler(NotFoundException.class)
    public ResponseEntity<ErrorMessageDto> handleNotFoundException(NotFoundException ex) {
        log.error(ex.getMessage());
        ErrorMessageDto errorMessage = new ErrorMessageDto(ex.getMessage());
        return new ResponseEntity<>(errorMessage, HttpStatus.NOT_FOUND);
    }
}

File: car-factory/src/main/kotlin/com/zuzex/carfactory/exception/GlobalExceptionHandler.kt
package com.zuzex.carfactory.exception

import com.zuzex.common.dto.ErrorMessageDto
import com.zuzex.common.exception.NotFoundException
import mu.KotlinLogging
import org.springframework.http.HttpStatus
import org.springframework.http.ResponseEntity
import org.springframework.web.bind.annotation.ExceptionHandler
import org.springframework.web.bind.annotation.RestControllerAdvice

private val log = KotlinLogging.logger {}

@RestControllerAdvice
class GlobalExceptionHandler {

    @ExceptionHandler(NotFoundException::class)
    fun handleNotFoundException(ex: NotFoundException): ResponseEntity<ErrorMessageDto> {
        log.error(ex.message)
        return ResponseEntity(ErrorMessageDto(ex.message), HttpStatus.NOT_FOUND)
    }
}
